src/sender.py

```diff
@@ -85,3 +90,24 @@ def consume_stream(self):
                     is_rendezvous,
                 )
         return (None, None, None, None, None)
+
+    def request(self, method, url, data=None):
+        if not self.jwt:
+            with open("config.json") as json_config:
+                config = json.load(json_config)
+            username = config["username"]
+            password = config["password"]
+            response = requests.get(AUTH_ENDPOINT, auth=(username, password))
+            if response.status_code == 403:
+                # Invalid Credentials
+                return response
+            self.jwt = response.headers["Authorization"]
+        auth_header = {"Authorization": self.jwt}
+        if method == "get":
+            response = requests.get(url, headers=auth_header)
+        else:
+            response = requests.post(url, json=data, headers=auth_header)
+        if response.status_code == 403:
+            # Set to None so next time we re-authenticate (token most likely expired)
+            self.jwt = None
+        return response
```

Comment:
This method is identical to the one in `receiver.py` right? We should start thinking about handling code duplication between these two.

Reply:
Agreed. There's quite a lot of duplication all over the place